### PR TITLE
Add function 'sd_excitations' to 'qchem' module

### PR DESCRIPTION
**Pull request by agran2018**
**Context:**
Implementing the Unitary Coupled-Cluster Singles Doubles (UCCSD) ansatz requires a function to generate all possible single and double excitations from the Hartree-Fock reference state.

**Description of the Change:**
Implement the new function ``sd_excitations`` in the ``qchem`` module.

**Benefits:**
This function can be generalized, if necessary, to generate higher-order excitations (triple, quadruple, etc.).

**Possible Drawbacks:**
None

**Related GitHub Issues:**
None

### CODE REVIEW

File: qchem/pennylane_qchem/qchem.py [hunk 1]
```diff
@@ -571,4 +652,5 @@ def generate_hamiltonian(
     "_qubit_operators_equivalent",
     "convert_hamiltonian",
     "generate_hamiltonian",
+    "sd_configs",
```
Comment (josh146) [inline qchem/pennylane_qchem/qchem.py:655]:
```suggestion
    "sd_excitations",
```

**Comment by agran2018** [comment]
> Thanks @agran2018!

Thanks @josh146 !.

File: qchem/pennylane_qchem/qchem.py [hunk 2]
```diff
@@ -561,6 +561,87 @@ def generate_hamiltonian(
...
+        )
+
+    if n_orbitals <= n_electrons:
+        raise ValueError(
+            "The number of active orbitals ({}) "
+            "has to be greater than the number of active electrons ({})."
+            .format(n_orbitals, n_electrons)
+        )
+
+    if int(delta_sz) not in (0, 1, -1, 2, -2):
```
Comment (soranjh) [inline qchem/pennylane_qchem/qchem.py:618]: Is the int necessary here?

File: qchem/pennylane_qchem/qchem.py [hunk 3]
```diff
@@ -561,6 +561,87 @@ def generate_hamiltonian(
...
+    >>> ph, pphh = sd_configs(2, 4, 0)
+    >>> print(ph)
+    [[0, 2], [1, 3]]
+    >>> print(pphh)
+    [[0, 1, 2, 3]]
+
+    Args: 
+        n_electrons (int): number of active electrons 
+        n_orbitals (int): number of active orbitals
+        delta_sz (int): Spin-projection selection rule.
```
Comment (soranjh) [inline qchem/pennylane_qchem/qchem.py:593]: Is the choice of the selection rule always straightforward? Is it possible to have a default value for it?

File: qchem/pennylane_qchem/qchem.py (hunk 2 above)
Comment (agran2018) [reply qchem/pennylane_qchem/qchem.py:618]: Thanks @soranjh for this point. It is not necessary. Indeed, having the ``int`` breaks this check:
```
if int(delta_sz) not in (0, 1, -1, 2, -2):
        raise ValueError(
            "Expected values for 'delta_sz' are 0, +/- 1 and +/- 2 but got ({})."
             .format(delta_sz)
        )
```
for, say, ``delta_sz=1.5``, which is not OK. Good catch!.

File: qchem/pennylane_qchem/qchem.py (hunk 3 above)
Comment (agran2018) [reply qchem/pennylane_qchem/qchem.py:593]: Yes, and I think it is convenient. The natural default is to stay in the spin sector of the HF reference state, that is, ``delta_sz=0``. 
I had it before but as we decided to code up UCCSD as template, it was gone.
Thanks @soranjh .

**Pull request merged**